container/kvm/initialisation.go

```diff
@@ -56,3 +84,117 @@ func ensureDependencies() error {
 
 	return nil
 }
+
+// createPool creates the libvirt storage pool directory. runCmd and chownFunc
+// are here for testing. runCmd so we can check the right shell out calls are
+// made, and chownFunc because we cannot chown unless we are root.
+func createPool(pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
```

Comment:
I can see that and I think if this were ever going to be used outside of this file -- more or less package -- it would be worthwhile to refactor. As it is, it should never be used outside of here, so an unexported type with unexported methods just means a bunch more more constructors and test code. As it is I don't see a cost-benefit worth changing it, is that OK with you?

findPath nee pathfinder fakes paths.DataDir which takes a string arg and returns an error:(  

Hopefully, one day soon digital ocean's libvirt and qemu libraries grow the functionality we need and this could all just go away:)